Infrastructure/Services/Repositories/UnitOfWork.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Application.Interfaces.IRepositories;
+using Infrastructure.Data;
+using Microsoft.Extensions.Logging;
+
+namespace Infrastructure.Services.Repositories;
+
+public class UnitOfWork : IUnitOfWork, IDisposable
```

Comment:
Think about using transactions, it's one of the goal of UnitOfWork.

Reply:
added transaction